java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java

```diff
@@ -223,6 +223,7 @@ public interface TemplateConstants {
   String SPARK_READ_FORMAT_BIGQUERY = "com.google.cloud.spark.bigquery";
   String INTERMEDIATE_FORMAT_OPTION_NAME = "intermediateFormat";
   String INTERMEDIATE_FORMAT_ORC = "orc";
+  String SPARK_SAVE_MODE_OVERWRITE = "overwrite";
```

Comment:
needs to be supplied via template.properties and command line args.

Reply:
While this is supplied by template properties in most cases, we use this constant value for a specific edge case when the source data lack partitions. 

When the source data lacks partitions the template will read all of the source data and upload to the target table. This must be an overwrite since we are not identifying new data, else the template would be uploading new and old data on every run thus generating duplicates.

Even though we could just rely on the user provided input value for `save mode` I believe in the specific case on non partitioned source data this input is prone to failure / misuse (user might forget to change the save mode). Also it seems like unnecessary work for the user, since they would have to manually identify which sources lack partitions and manually correct the command line args for it. This is an unnecessary task for the user since the template is already aware when a source lacks partitions and can easily set the proper value of `save mode` on behalf of the user

Reply:
ok I guess we can move forward with this and change later, if needed